### ISSUE TITLE
YAML formatting restrictions

### ISSUE THREAD

**Issue by leobalter** [tc39/test262#1997]
We should restrict the YAML format for the test files frontmatter to improve interoperability.

Many test runners depend on quick but limited yaml parsers and it's ok to have style restrictions identified in the importing processes for Test262.

Some of them I can already tell from scratch:

- [ ] no leading or trailing spaces from inline lists. e.g.: `[ leading-space, foo, trailing-space ]`
- [ ] no leading or trailing blank lines from text blocks. e.g. the info tag cannot start with an empty line
- [ ] lists should always be inline. Always wrapped with square brackets and comma-space divided words. e.g.:

```diff
features:
-  - var
-  - foo
-  - bar
+  [var, foo, bar]
```

There are might be more items we want to identify here. We should increase the list as necessary.

cc @caitp @gsathya @psmarshall

**Comment by gsathya** [comment]
Looks like JSC's parser is having trouble with the YAML as well: https://bugs.webkit.org/show_bug.cgi?id=193053

**Comment by leobalter** [comment]
@gsathya yes, the issues we can list here might affect at least V8, JSC and SpiderMonkey. Turns out it Yaml parsers are more challenging than one can expect.

**Comment by gsathya** [comment]
Another one: https://github.com/tc39/test262/issues/1997

**Comment by littledan** [comment]
Thanks for working on this important issue, @leobalter.

**Comment by kmiller68** [comment]
Would it also be possible to prohibit inline value strings that span multiple lines? That seems to cause issues for JSC's YAML parser... 

For example:
```
description: Security Throws a TypeError if start coercion detaches
  the buffer
```

Ideally, this would be formatted as one line or with "|"/">"

**Comment by rkirsling** [comment]
Opened #2505 to address the specific issue @kmiller68 identified.
